internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1347,6 +1348,30 @@ func TestMachineSetReconciler_syncMachines(t *testing.T) {
 		g.Expect(updatedDeletingMachine.Spec.NodeDeletionTimeout).Should(Equal(deletingMachine.Spec.NodeDeletionTimeout))
 		g.Expect(updatedDeletingMachine.Spec.NodeVolumeDetachTimeout).Should(Equal(deletingMachine.Spec.NodeVolumeDetachTimeout))
 	}, 5*time.Second).Should(Succeed())
+
+	// Verify in-place mutable fields are updated on the deleting machine
+	ms.Spec.Template.Spec.NodeDrainTimeout = duration11s
+	ms.Spec.Template.Spec.NodeDeletionTimeout = duration11s
+	ms.Spec.Template.Spec.NodeVolumeDetachTimeout = duration11s
+	g.Expect(reconciler.syncMachines(ctx, ms, []*clusterv1.Machine{updatedInPlaceMutatingMachine, deletingMachine})).To(Succeed())
+	updatedDeletingMachine := deletingMachine.DeepCopy()
+
+	g.Eventually(func(g Gomega) {
```

Comment:
Eventually shouldn't be required here (syncMachines does synchronous calls and GetAPIReader() reads without cache)

Reply:
fixed